datastore/schema.sql

```diff
@@ -39,16 +39,32 @@ create table if not exists "rbac-policies" (
 	constraint "rbac-policies.check-name" check (name <> '')
 );
 
+create table if not exists permissions (
+	_id text not null check (_id <> ''),
+
+	constraint "permissions.pkey" primary key (_id)
+);
+
```

Comment:
I don't think we need a separate table for permissions since it's just a single field in here. Instead we can simply store the permission in `roles_permissions` table (`role_id`, `permission`).

However, I can see having a table will allow us to make permissions unique across all roles and add gRPC endpoints to manage permissions but this would mean permissions have to be created before creating roles (and only allow adding from the pre-existing list) with the benefit of not allowing the same permission across multiple roles accidentally. This will also mean current behaviour of the gRPC endpoint to create roles will need to change to throw an error if permissions doesn't exist. 

@neculalaura thoughts?

Reply:
If you don't add any other fields, then yes, any data you can get from the `permissions` table you can also get from the `roles_permissions` table. The only advantage that I see is that you can ensure unique permissions across roles, as you said. The other advantage is that the `permissions` table is easily expandable in the future in case you want to add any extra info.

> This will also mean current behaviour of the gRPC endpoint to create roles will need to change to throw an error if permissions doesn't exist.

you can add logic to create the permission if it doesn't exist. This way you sort of abstract the permission layer and save users from making an extra call

Reply:
@td0m think it's fine to use a table for `permissions` as you've proposed 👍.

Reply:
But for the gRPC endpoints, can we add endpoints to manage (at least create) permissions and separate out adding permissions to roles as a separate request (i.e. a role will need to be created first and then add permissions) please?